csp_billing_adapter/csp_hookspecs.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright 2023 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pluggy
+
+from csp_billing_adapter.config import Config
+
+hookspec = pluggy.HookspecMarker('csp_billing_adapter')
+
+
+@hookspec(firstresult=True)
+def meter_billing(
+    config: Config,
+    dimensions: dict,
+    timestamp: str,
+):
+    """
+    Process metering request against the CSP API.
+
+    :param config: The application configuration dictionary
+    :param dimensions: A hash of the billing dimensions and quantities
```

Comment:
Aren't the dimensions part of the adapter config?

Reply:
Yep. And there's a separate mapping of dimensions to usage for metering a bill.